slatedb/src/db_snapshot.rs

```diff
@@ -121,6 +121,7 @@ impl DbSnapshot {
                 &db_state,
                 None,
                 Some(self.started_seq),
+                None,
```

Comment:
Just had a random thought: ScanOptions allows you to set `dirty=true`. But if we call `scan`, do we want to allow this? 🤔 I suppose it's technically possible to do a dirty scan with a seqnum that hasn't yet been committed? 🤔 Not directly related to this PR--just thinking out loud.

Reply:
after some thoughts, I think this is technically feasible 😲, this might potentially lead to a higher incidence of conflict errors during transaction conflict checks since it collects more keys for the read-write conflict check, but it should not affect transactional safety i think. 🤔